### PR TITLE
Modified to use a share report nuget package

### PR DESCRIPTION
**Pull request by jpalac**
Added the usage of Particular.LicensingComponent.Report and removed duplicated code and tests

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -22,10 +20,10 @@
     <PackageVersion Include="NUnit.Analyzers" Version="4.2.0" />
     <PackageVersion Include="NUnit3TestAdapter" Version="4.5.0" />
     <PackageVersion Include="Particular.Approvals" Version="1.0.0" />
+    <PackageVersion Include="Particular.LicensingComponent.Report" Version="1.0.0-alpha.2" />
```
Comment (johnsimons) [inline src/Directory.Packages.props:23]: it seems the package is still alpha, are planning to release the package first?

File: src/AppCommon/AppCommon.csproj [hunk 2]
```diff
@@ -1,24 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (johnsimons) [inline src/AppCommon/AppCommon.csproj:4]: So we are also updating to net8, which means we also need to update https://docs.particular.net/nservicebus/throughput-tool/#installation-net-tool-recommended

**Review by johnsimons — APPROVED** [review]
Overall seems fine.
But i left a few comments that need addressing before merging

File: src/AppCommon/AppCommon.csproj (hunk 2 above)
Comment (jpalac) [reply src/AppCommon/AppCommon.csproj:4]: yep - https://github.com/Particular/docs.particular.net/pull/6701

File: src/Directory.Packages.props (hunk 1 above)
Comment (jpalac) [reply src/Directory.Packages.props]: yep did that this morning

**Review by abparticular — APPROVED** [review]

**Pull request merged**